OpenSim/Common/ComponentSocket.h

```diff
@@ -721,6 +721,19 @@ class Input : public AbstractInput {
         SimTK_INDEXCHECK_ALWAYS(index, getNumConnectees(),
                                 "Input<T>::setAlias()");
 
+        const auto& connecteeName = getConnecteeName(index);
+        std::string ignore1{}, ignore2{}, ignore3{};
+        std::string currAlias{};
+        parseConnecteeName(connecteeName,
+                           ignore1, ignore2, ignore3, currAlias);
+        if(currAlias.empty()) {
+            setConnecteeName(getConnecteeName(index) + "(" + alias + ")",
```

Comment:
Done.
I have added a function to compose the path. The fix uses it to set the alias.